### PR TITLE
Feature/d3 asim 2859

### PR DESCRIPTION
**Pull request by faizan2590**

### CODE REVIEW

File: src/d3a/models/strategy/market_maker_strategy.py
Comment (hannesdiedrich) [inline src/d3a/models/strategy/market_maker_strategy.py:49]: Just out of curiosity: What other type it could be?

File: src/d3a/d3a_core/live_events.py
Comment (hannesdiedrich) [inline src/d3a/d3a_core/live_events.py:65]: Can you explain why this is needed? This reminds me actually, to remove these already in the d3a-web.
But even now: if there is no "number_of_clones" member, this will fail, so i suggest to remove this line, if you can live without it (and maybe create a small PR that deals with this in the d3a-web) Thanks

**Review by hannesdiedrich — APPROVED** [review]
Two comments, LGTM once addressed

File: src/d3a/models/strategy/market_maker_strategy.py
Comment (faizan2590) [reply src/d3a/models/strategy/market_maker_strategy.py:49]: `grid_connected ` must always be boolean.

File: src/d3a/models/area/__init__.py
Comment (spyrostz) [inline src/d3a/models/area/__init__.py:140]: Aren't these already done by the `activate` method https://github.com/gridsingularity/d3a/blob/master/src/d3a/models/area/__init__.py#L232 ?

File: src/d3a/d3a_core/live_events.py
Comment (spyrostz) [reply src/d3a/d3a_core/live_events.py:65]: Agree with @hannesdiedrich , and in order to circumvent the KeyError here, you can use the pop method: 
`self.area_params.pop('number_of_clones', None)`

File: src/d3a/models/strategy/market_maker_strategy.py
Comment (spyrostz) [reply src/d3a/models/strategy/market_maker_strategy.py:49]: That reminds me, at some point we should think about using the scenario validator for live events (or do we use it even now ? I cannot recall). Should be dealt on its own story though

File: src/d3a/d3a_core/live_events.py
Comment (spyrostz) [inline src/d3a/d3a_core/live_events.py:55]: This line can be deleted. type is also removed from the dict couple lines below

**Review by spyrostz — COMMENTED** [review]
Some comments, LGTM once resolved

File: src/d3a/models/strategy/market_maker_strategy.py
Comment (hannesdiedrich) [reply src/d3a/models/strategy/market_maker_strategy.py:49]: You are right, we dont validate the live events with the scenario validator. shall i add a story?

File: src/d3a/models/area/__init__.py
Comment (faizan2590) [reply src/d3a/models/area/__init__.py]: Yes, tested it and you are right.

File: src/d3a/d3a_core/live_events.py
Comment (faizan2590) [reply src/d3a/d3a_core/live_events.py:57]: Actually I have deleted from there and have to keep it here else the new strategy initialisation would complain as `type` parameter is unknown to it and ATM strategy doesn't accept arbitrary args and kwargs either.

File: src/d3a/models/strategy/market_maker_strategy.py
Comment (faizan2590) [reply src/d3a/models/strategy/market_maker_strategy.py:49]: Done: https://gridsingularity.atlassian.net/browse/D3ASIM-2961

File: src/d3a/d3a_core/d3a_jobs.py
Comment (hannesdiedrich) [inline src/d3a/d3a_core/d3a_jobs.py:44]: please change before merging

File: src/d3a/d3a_core/live_events.py
Comment (hannesdiedrich) [reply src/d3a/d3a_core/live_events.py:57]: Why are you handling `type` differently from e.g. `energy_rate_profile_uuid ` (meaning the difference between `del` and `pop`?

File: src/d3a/models/area/__init__.py
Comment (hannesdiedrich) [inline src/d3a/models/area/__init__.py:141]: Please delete

File: src/d3a/models/strategy/market_maker_strategy.py
Comment (hannesdiedrich) [reply src/d3a/models/strategy/market_maker_strategy.py:49]: Thanks for adding the ticket, still, the check for boolean can be removed.

File: src/d3a/d3a_core/live_events.py
Comment (hannesdiedrich) [reply src/d3a/d3a_core/live_events.py]: Could you please add another story to deprecate `number_of_clones` in d3a_web (at least the sending of this parameter to the d3a, which was never needed). Thanks!

**Review by hannesdiedrich — DISMISSED** [review]
Some more comments, once addressed, LGTM

File: src/d3a/d3a_core/live_events.py
Comment (spyrostz) [reply src/d3a/d3a_core/live_events.py:57]: Also, if you are using pop, there could be only one place where all these parameters get deleted (exactly after `if key_in_dict_and_not_none(self.area_params, 'type'):`). There could even be a small function called, for instance, sanitize_market_maker_dict that could remove all these parameters from the dict regardless of the type value. Then you could also keep the type as a variable, and use this for the if/else blocks.

File: src/d3a/d3a_core/live_events.py
Comment (spyrostz) [inline src/d3a/d3a_core/live_events.py:62]: You can also construct these objects without assigning the class object first if I understand the code correctly.

File: src/d3a/models/area/__init__.py
Comment (spyrostz) [inline src/d3a/models/area/__init__.py:143]: I think you can also move all these lines to the event too. I am mentioning this in order to leave the `area_reconfigure_event` completely unaffected from the live events.

**Review by spyrostz — APPROVED** [review]
Some cosmetic/code organization comments, LGTM however once these are discussed.

File: src/d3a/models/strategy/market_maker_strategy.py
Comment (faizan2590) [reply src/d3a/models/strategy/market_maker_strategy.py:49]: I would rather be cautious and keep it here till the validators have been added.

File: src/d3a/d3a_core/live_events.py
Comment (faizan2590) [reply src/d3a/d3a_core/live_events.py]: https://gridsingularity.atlassian.net/browse/D3ASIM-2989
Done!

File: src/d3a/d3a_core/live_events.py
Comment (spyrostz) [inline src/d3a/d3a_core/live_events.py:54]: you could have a clause `area_type = self.area_params.pop("type")`. That way you will avoid the explicit del's below

**Review by spyrostz — APPROVED** [review]
One small comment, LGTM though

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Review by spyrostz — APPROVED** [review]
LGTM

**Pull request merged**
